src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, DrawGenerator {
```

Comment:
It seems that you have used an abstract class to set behavior for classes. Consider whether an interface or a regular class would be more appropriate for your needs. Abstract classes are typically used when there is shared code among all subclasses, but if you're just specifying a contract for subclasses to follow, an interface might be a better choice.